src/libstd/panicking.rs

```diff
@@ -38,6 +39,7 @@ enum Handler {
 
 static HANDLER_LOCK: StaticRwLock = StaticRwLock::new();
 static mut HANDLER: Handler = Handler::Default;
+static mut FIRST_PANIC: AtomicBool = AtomicBool::new(true);
```

Comment:
Oh, right! I would have thought that `compare_and_swap` takes a mutable reference...